README.md

```diff
@@ -463,4 +463,47 @@ def ==(other)
   other.is_a?(Quantity) && ...
   # ...or, sometimes, duck type
   other.respond_to?(:amount) && other.respond_to?(:unit) && ...
+end
+```
+
+## Pattern Matching
+
+Value objects are a prime candidate for use with the pattern matching syntax introduced in Ruby 2.7.
+
+The easiest way to integrate with pattern matching is to return a hash of attributes from `#decontruct_keys`, which you have probably already implemented in `#to_h`. This allows the object to be pattern matched using the hash syntax:
+
+```ruby
+class Quantity
+  ...
+
+  def decontruct_keys(_keys)
+    to_h
+  end
+end
+
+case Quantity.new(10, 'm')
+in { unit: 'm', amount: }
+  puts "Metric: #{amount} meters"
+in { unit: 'ft', amount: }
+  puts "Imperial: #{amount} feet"
+end
+```
+
+Consider also supporting the array syntax if the attributes have an obvious or intuitive ordering (See also: keyword arguments vs positional arguments in [Construction](#construction)). This is done by returning an array of attributes from the `#deconstruct` method:
+
+```ruby
+class Quantity
+  ...
+
+  def deconstruct
+    [@amount, @unit]
+  end
+end
+
+case Quantity.new(10, 'm')
+in [amount, 'm']
```

Comment:
Again, maybe no brackets? I feel like it is more expressive (YMMV)

Reply:
Same as above.